pysidt/sidt.py

```diff
@@ -310,26 +311,93 @@ def evaluate(self, mol):
         return node.rule
 
 
+def to_dict(obj):
+    out_dict = dict()
+    out_dict["class"] = obj.__class__.__name__
+    attrs = [attr for attr in dir(obj) if not attr.startswith("_")]
+    for attr in attrs:
+        val = getattr(obj, attr)
+
+        if callable(val) or val == getattr(obj.__class__, attr):
+            continue
+
+        try:
+            json.dumps(val)
+            out_dict[attr] = val
+        except:
+            if isinstance(val, ScalarQuantity):
+                d = {
```

Comment:
Is this right? `d` is not used anywhere? Can we add test for this function?

Reply:
Fixed, we do test this in the notebook, although it's more of a functional test through write_nodes. I can construct a nosetests framework. I'm not very familiar with pytest though, so if we want to continue using pytest it's probably most efficient if you setup the framework.